test/unit/org/apache/cassandra/io/util/ThreadLocalReadAheadBufferTest.java

```diff
@@ -139,10 +139,9 @@ private Gen<InputData> lastBlockReads()
 
     private Gen<InputData> randomReads()
```

Comment:
the generator after the change looks identical to lastBlockReads()..

it also looks like the naming is mixed up:  randomReads was about reading the last block and lastBlockReads is about reading from anywhere.

should we preserve reading the test to read the last block by using max(0, fileSize(file) - blockSize)?

Reply:
That was the other way I was looking at fixing it but was having a hard time piecing back the context. Ill test this change now